org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -312,30 +311,36 @@ private static void updateMarkers(Map<IResource, List<Issue>> issuesPerResource,
         continue;
       }
       if (resource instanceof IFile) {
-        List<Issue> rawIssues = resourceEntry.getValue();
-        IPath path = resource.getFullPath();
-        try {
-          textFileBufferManager.connect(path, LocationKind.IFILE, new NullProgressMonitor());
-          ITextFileBuffer textFileBuffer = textFileBufferManager.getTextFileBuffer(path, LocationKind.IFILE);
-          IDocument document = textFileBuffer.getDocument();
-
-          IProject project = resource.getProject();
-          String moduleKey = project.getName();
-          SonarLintCorePlugin.getDefault().getModulePathManager().setModulePath(moduleKey, project.getLocation().toString());
-
-          String relativePath = resource.getProjectRelativePath().toString();
-          trackLocalIssues(moduleKey, relativePath, document, rawIssues);
-          trackServerIssues(moduleKey, relativePath, resource);
-        } finally {
-          textFileBufferManager.disconnect(path, LocationKind.IFILE, new NullProgressMonitor());
-        }
+        trackIssues(resource, textFileBufferManager, resourceEntry.getValue(), triggerType);
       } else {
         // TODO handle non-file-level issues
       }
     }
   }
 
-  private static void trackLocalIssues(String moduleKey, String relativePath, @Nullable IDocument document, List<Issue> rawIssues) {
+  private void trackIssues(IResource resource, ITextFileBufferManager textFileBufferManager, List<Issue> rawIssues, TriggerType triggerType) throws CoreException {
+    IPath path = resource.getFullPath();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this unused "path" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1481)